src/node_crypto.cc

```diff
@@ -531,6 +531,24 @@ void SecureContext::Initialize(Environment* env, Local<Object> target) {
   env->set_secure_context_constructor_template(t);
 }
 
+SecureContext::SecureContext(Environment* env, v8::Local<v8::Object> wrap)
+    : BaseObject(env, wrap) {
+  MakeWeak();
+  env->isolate()->AdjustAmountOfExternalAllocatedMemory(kExternalSize);
+}
+
+inline void SecureContext::Reset() {
+  if (ctx_ != nullptr) {
+    env()->isolate()->AdjustAmountOfExternalAllocatedMemory(-kExternalSize);
+  }
+  ctx_.reset();
```

Comment:
Shouldn't this be in the `!= nullptr` check?

Reply:
It does not need to be. IIRC it's a smart pointer here. The `!= nullptr` is using overridden `!=` operator and the `.reset()` is a non-op if the `ctx_` is not set.